packages/core/src/blocks/shaderBlock.ts

```diff
@@ -121,25 +123,38 @@ export abstract class ShaderBlock extends DisableableBlock {
         runtime.registerResource(shaderBlockRuntime);
 
         if (finalOutput) {
-            runtime.registerCommand(
-                createCommand(`${this.getClassName()}.renderToCanvas`, this, () => {
-                    shaderBlockRuntime.renderToCanvas();
-                })
-            );
-        } else {
-            const rtt =
-                this.output.runtimeData && (this.output.runtimeData.value as ThinRenderTargetTexture).renderTarget;
-            if (!rtt) {
-                throw new Error("ShaderBlock does not have a render target texture.");
+            if (initializationData.outputBlock.renderTargetTexture) {
```

Comment:
Any reason to factor as a helper with the outputBlock code ?

Reply:
Good suggestion - done!